libraries/Microsoft.Bot.Builder.Core/ITurnContext.cs

```diff
@@ -23,7 +23,7 @@ namespace Microsoft.Bot.Builder
     /// <seealso cref="BotAdapter"/>
     /// <seealso cref="UpdateActivityHandler"/>
     /// <seealso cref="DeleteActivityHandler"/>
-    public delegate Task SendActivitiesHandler(ITurnContext context, List<Activity> activities, Func<Task> next);
+    public delegate Task<ResourceResponse[]> SendActivitiesHandler(ITurnContext context, List<Activity> activities, Func<Task<ResourceResponse[]>> next);
```

Comment:
Huh. I would have bet money I had already done this, and had tests around it. 

Thanks for wiring it up.